service/api/src/test/java/org/eclipse/kapua/KapuaExceptionTest.java

```diff
@@ -12,37 +12,293 @@
+            }
+            try {
+                kapuaException.getLocalizedMessage();
+                fail("NullPointerException expected.");
+            } catch (Exception e) {
+                assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+            }
+        }
+    }
 
-        ke = KapuaException.internalError(new NullPointerException());
-        Assert.assertEquals("An internal error occurred: java.lang.NullPointerException.", ke.getMessage());
+    @Test
+    public void internalErrorCauseMessageTest() {
+        Throwable[] cause = {new Throwable(), null};
+        String[] messages = {"Message", null};
 
-        ke = new KapuaException((KapuaErrorCode) () -> "MISSING", "abc", 1);
-        Assert.assertEquals("Error: abc, 1", ke.getMessage());
+        for (Throwable throwable : cause) {
+            for (String msg : messages) {
+                assertThat("Instance of KapuaException expected.", KapuaException.internalError(throwable, msg), IsInstanceOf.instanceOf(KapuaException.class));
+                assertEquals("Expected and actual values should be the same.", new KapuaException(KapuaErrorCodes.INTERNAL_ERROR, throwable, msg).toString(), KapuaException.internalError(throwable, msg).toString());
+                assertEquals("Expected and actual values should be the same.", "An internal error occurred: " + msg + ".", KapuaException.internalError(throwable, msg).getMessage());
+                assertEquals("Expected and actual values should be the same.", "An internal error occurred: " + msg + ".", KapuaException.internalError(throwable, msg).getLocalizedMessage());
+                assertEquals("Expected and actual values should be the same.", throwable, KapuaException.internalError(throwable, msg).getCause());
+                assertEquals("Expected and actual values should be the same.", KapuaErrorCodes.INTERNAL_ERROR, KapuaException.internalError(throwable, msg).getCode());
+                assertEquals("Expected and actual values should be the same.", expectedErrorMessage, KapuaException.internalError(throwable, msg).getKapuaErrorMessagesBundle());
+            }
+        }
     }
 
     @Test
-    public void missingBundleTest() {
-        KapuaExceptionWithoutBundle exceptionWithoutBundle = new KapuaExceptionWithoutBundle(KapuaErrorCodes.ILLEGAL_STATE, "param1", "param2");
-        Assert.assertEquals("Error: param1, param2", exceptionWithoutBundle.getMessage());
+    public void internalErrorCauseTest() {
+        String message = "Message";
+        Throwable nullThrowable = null;
+        Throwable[] throwables = {new Throwable(message), new Throwable()};
+        String[] expectedMessage = {"An internal error occurred: " + message + ".", "An internal error occurred: " + throwables[1] + "."};
+        String[] arguments = {"Message", "java.lang.Throwable"};
+
+        for (int i = 0; i < throwables.length; i++) {
```

Comment:
Enhanced for loop? 
        for (Throwable throwableParameter: throwables) {

Reply:
I used index i to get values from different arrays in the same position